src/mem/snoop_filter.cc

```
@@ -132,6 +132,12 @@ SnoopFilter::lookupRequest(const Packet* cpkt, const ResponsePort&
             // to the CPU, already -> the response will not be seen by this
             // filter -> we do not need to keep the in-flight request, but make
             // sure that we know that that cluster has a copy
+            SimObject * temp_simobject =find("system");
+            System * temp_system = dynamic_cast<System *> (temp_simobject);
+            if(temp_system->restore_from_memtrace_enabled() == true)
+            {
```

Cppcheck (2.19.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)